RTN10c

ricardopereira opened the pull request.

ricardopereira commented:
Not passing but I think I implemented it well.

mattheworiordan commented:
LGTM

tcard commented on ablySpec/RealtimeClient.connection.swift:
```diff
@@ -713,6 +713,44 @@ class RealtimeClientConnection: QuickSpec {
+                    for _ in 1...5 {
+                        channel.publish("message", cb: { status in
+                            expect(status.state).to(equal(ARTState.Ok))
+                            lastSerial = client.connection().serial
```
Correct me if I'm wrong, but there's no guarantee that the last callback called will be that of the last published, since it could take the server more time to respond to any of them. (I've checked the client code and that seems to be indeed what's happening.)

tcard replied:
Sorry, scratch that, I though that connection.serial was updated before publishing, but no, it is set when the N?ACK message is received.

tcard commented:
The test looks good to me.

mattheworiordan commented on ablySpec/RealtimeClient.connection.swift:
```diff
@@ -713,6 +713,44 @@ class RealtimeClientConnection: QuickSpec {
+                    }
+                    expect(lastSerial).toEventually(equal(4), timeout: testTimeout)
+
+                    let options = AblyTests.clientOptions(requestToken: true)
```
Why are you using token auth here?  Not sure how that applies to this test when you have not used token auth for the first `client`

ricardopereira replied:
No reason in particular.

mattheworiordan replied:
I would remove as it's just more noise

ricardopereira replied:
Done

mattheworiordan commented:
Mostly good, one small comment re: token auth

ricardopereira commented:
Rebased with no changes.

> Not passing but I think I implemented it well. The connectionSerial from the recovered client is 0.

Should I open an issue and merge?
